fcrepo4/fcrepo4
# Create (via PUT) ACL resources.

https://jira.duraspace.org/browse/FCREPO-2756

Create ACL resources via PUT.

# What's new?
Create ACL resources via PUT on the static fcr:acl URI http://resource/path/fcr:acl, and the ACL is created in container fedora:acl under the resource like TimeMap instead of linking with the acl:accessControl predicate.

# How should this be tested?
```
1. Create RDFSource
$ curl -i -XPOST -H "Slug: TestObject" -H "Content-Type: text/n3" --data-binary "<> <info:fedora/test/field> \"Test Object\"" http://localhost:8080/rest/

2. Create ACL on the RDFSource and verify status 201 with the location header URI translated to fcr:acl
$ curl -i -XPUT http://localhost:8080/rest/TestObject/fcr:acl
HTTP/1.1 201 Created
Date: Mon, 23 Apr 2018 14:35:18 GMT
ETag: W/"75a70600c3237bd40744b79377f218d444fae0eb"
Last-Modified: Mon, 23 Apr 2018 14:35:18 GMT
Link: <http://localhost:8080/static/constraints/ContainerConstraints.rdf>; rel="http://www.w3.org/ns/ldp#constrainedBy"
Location: http://localhost:8080/rest/TestObject/fcr:acl
Content-Type: text/plain
Content-Length: 45
Server: Jetty(9.2.3.v20140905)

3. Create NonRdfSource
$ curl -i -XPOST -H "Slug: binary"  http://localhost:8080/rest -H "Content-Type: application/octet-stream" --data-binary "binary content"

4. Create ACL on the NonRdfSource and verify status 201 with the location header URI translated to fcr:acl
$ curl -i -XPUT http://localhost:8080/rest/binary/fcr:acl
HTTP/1.1 201 Created
Date: Mon, 23 Apr 2018 14:50:41 GMT
ETag: W/"cb84ff95041ef51503db7a6b6b01e9db2df1ddd2"
Last-Modified: Mon, 23 Apr 2018 14:50:41 GMT
Link: <http://localhost:8080/static/constraints/ContainerConstraints.rdf>; rel="http://www.w3.org/ns/ldp#constrainedBy"
Location: http://localhost:8080/rest/binary/fcr:acl
Content-Type: text/plain
Content-Length: 41
Server: Jetty(9.2.3.v20140905)
```
